src/deepMap.ts

```diff
@@ -1,65 +1,95 @@
-/**
- * @private
- */
+// Registered Symbols are not supported as keys in a WeakMap.
+// Firefox also does not yet support any Symbol as a WeakMap key.
+function supportedInWeakMap(symbol: Symbol) {
+    try {
+        new WeakMap().set(symbol, undefined)
+        return true
+    } catch {
+        return false
+    }
+}
+
 export class DeepMapEntry<T> {
-    private root: Map<any, any>
-    private closest: Map<any, any>
-    private closestIdx: number = 0
     isDisposed = false
+    private hash: string
+
+    constructor(
+        private readonly state: {
+            store: Map<string, T>
+            weakHashes: WeakMap<object | Function | Symbol, string> | undefined
+            strongHashes: Map<Symbol, string> | undefined
```

Comment:
An alternate method (if lazily evaluating them has implications on map files) could be to detach them from the instance and use a global instead. This will also reduce the memory requirements of `state` and may afford de-duplication efficiency (if two `computedFn`s reference use the same object argument).

```ts
const weakHashes = new WeakMap<object | Function | Symbol, string>();
const strongHashes = new Map<Symbol, string>();

function prehash(...) { ... }
function getHash(...) { ... }
```